include/ftl/iterator_interface.hpp

```diff
@@ -468,6 +472,13 @@ class const_iterator_interface {
     return static_cast<const Derived &>(*this).preincrement_impl();
   }
 
+    /**
+   * Advances the iterator.
+   */
+  auto operator++(int) const -> decltype(auto) {
```

Comment:
Why was this added? Why it was only added to const_iterator_interface and not iterator_interface?